src/time/duration-panel.vue

```diff
@@ -0,0 +1,163 @@
+<template>
+  <div :class="`${prefixClass}-time`">
+    <div v-if="showDurationHeader" :class="`${prefixClass}-time-header`">
+      <button
+        type="button"
+        :class="`${prefixClass}-btn ${prefixClass}-btn-text ${prefixClass}-time-header-title`"
+        @click="handleClickTitle"
+      >
+        {{ title }}
+      </button>
+    </div>
+    <div :class="`${prefixClass}-time-content`">
+      <list-columns
+        :date="innerValue"
+        :get-classes="getClasses"
+        :hour-options="hourOptions"
+        :minute-options="minuteOptions"
+        :second-options="secondOptions"
+        :hour-step="hourStep"
+        :minute-step="minuteStep"
+        :second-step="secondStep"
+        v-bind="ShowHourMinuteSecondAMPM"
+        @select="handleSelect"
+      ></list-columns>
+    </div>
+  </div>
+</template>
+
+<script>
+import { format } from 'date-format-parse';
+import { getValidDate } from '../util/date';
+import ListColumns from './list-columns';
+import { getLocaleFieldValue } from '../locale';
+
+export default {
+  name: 'DurationPanel',
+  components: { ListColumns },
+  inject: {
+    t: {
+      default: () => getLocaleFieldValue,
+    },
+    prefixClass: {
+      default: 'mx',
+    },
+  },
+  props: {
+    value: {},
+    defaultValue: {
+      default() {
+        const date = new Date();
+        date.setHours(0, 0, 0, 0);
+        return date;
+      },
+    },
+    format: {
+      default: 'DD HH',
+    },
+    timeTitleFormat: {
+      type: String,
+      default: 'YYYY-MM-DD',
+    },
+    showDurationHeader: {
+      type: Boolean,
+      default: false,
+    },
+    disabledDate: {
+      type: Function,
+      default: () => false,
+    },
+    dayOptions: Array,
+    hourOptions: Array,
+    minuteOptions: Array,
+    secondOptions: Array,
+    dayStep: {
+      type: Number,
+      default: 1,
+    },
+    hourStep: {
+      type: Number,
+      default: 1,
+    },
+    minuteStep: {
+      type: Number,
+      default: 1,
+    },
+    secondStep: {
+      type: Number,
+      default: 1,
+    },
+    showDay: {
+      type: Boolean,
+      default: undefined,
+    },
+    showHour: {
+      type: Boolean,
+      default: undefined,
+    },
+    showMinute: {
+      type: Boolean,
+      default: undefined,
+    },
+    showSecond: {
+      type: Boolean,
+      default: undefined,
+    },
+  },
+  computed: {
+    innerValue() {
+      return getValidDate(this.value, this.defaultValue);
+    },
+    title() {
+      const titleFormat = this.timeTitleFormat;
+      const date = new Date(this.innerValue);
+      return this.formatDate(date, titleFormat);
```

Comment:
@Xanngo I think these 3 lines above really could be written as a one line without vars declaring